packages/graphql/src/translate/queryAST/ast/fields/aggregation-fields/CountField.ts

```diff
@@ -43,6 +55,20 @@ export class CountField extends AggregationField {
     }
 
     public getAggregationProjection(target: Cypher.Variable, returnVar: Cypher.Variable): Cypher.Clause {
-        return new Cypher.Return([this.getAggregationExpr(target), returnVar]);
+        const resultMap = new Cypher.Map();
+
+        if (this.countFields.nodes) {
+            resultMap.set("nodes", this.getAggregationExpr(target));
+        }
+        if (this.countFields.edges) {
+            if (!this.edgeVar) {
+                throw new Error(
+                    "Edge variable not defined in Count field. This is likely a bug with the GraphQL library."
+                );
+            }
+            resultMap.set("edges", this.getAggregationExpr(this.edgeVar));
```

Comment:
Independently from the comment above, this `edgeVar` variable seems not needed. I tested the below, and the whole aggregation test suite works fine. This is because the client already invokes the `getAggregationProjection` with the correct variable depending on the context.
```suggestion
        if (this.countFields.edges) {
            resultMap.set("edges", this.getAggregationExpr(target));
```

Reply:
I think this may actually be a gap in the tests. As we don't yet support multiple relationships between 2 nodes, all the tests are covering a case where the count of nodes is the same as relationships, so these pass. 

With change you are proposing, the aggregation count cypher looks like this:

```cypher
MATCH (this)<-[this0:ACTED_IN]-(this1:cBfniRfaActor)
RETURN { nodes: count(this1), edges: count(this1) } AS var2
```

Reply:
You're right, I didn't notice it, for the count fields the target is always passed as "node" in the "AggregationOperation" while in `nodeFields` and `edgeFields` the target depends on the context!